deployment/address_book.go

```diff
@@ -82,14 +85,16 @@ type AddressBook interface {
 	AddressesForChain(chain uint64) (map[string]TypeAndVersion, error)
 	// Allows for merging address books (e.g. new deployments with existing ones)
 	Merge(other AddressBook) error
+	Remove(ab AddressBook) error
 }
 
 type AddressBookMap struct {
 	AddressesByChain map[uint64]map[string]TypeAndVersion
+	mtx              sync.RWMutex
```

Comment:
This seems reasonable, though do we want to consider just making the underlying map `sync.Map`? I'm not sure if we have operations that need to be atomic at the API edge, and if so, then obviously we need something more articulate like a RW lock. But would a threadsafe map be enough here?

Reply:
Yeah, sync.Map designed to solve a specific problem, the bottleneck on multicore systems. Long read here :) https://medium.com/@deckarep/the-new-kid-in-town-gos-sync-map-de24a6bf7c2c 

For current case, classic go approach is using sync.Mutex + map

Reply:
Got it. Makes sense. I certainly was thinking about it the way I think of ConcurrentHashMap. Good context links, thanks!